source/extensions/filters/common/ext_authz/check_request_utils.cc

```diff
@@ -199,6 +199,8 @@ void CheckRequestUtils::createHttpCheck(
   // *cb->connection(), callbacks->streamInfo() and callbacks->decodingBuffer() are not qualified as
   // const.
   auto* cb = const_cast<Envoy::Http::StreamDecoderFilterCallbacks*>(callbacks);
+
+  const std::string server_name(cb->connection()->ssl()->sni());
```

Comment:
Probably obviously there is no guarantee that this is a TLS connection so that has to be handled in the final patch.

Reply:
Of course! Fixed this up, I've also added an opt-in flag to limit the performance impact.